### PR TITLE
return changed for group_by

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

fixed the use of incorrect functions for lookup
made sure we return the correct 'changed' status

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

group_by
strategy

##### ANSIBLE VERSION

```
2.4/2.5
```

### CODE REVIEW

**Comment by tedder** [comment]
~~Looks like this may fix #32152, so backporting to 2.4 would be nice.~~

Doesn't fix that issue.

File: lib/ansible/plugins/strategy/__init__.py [hunk 1]
```diff
@@ -633,6 +633,7 @@ def _add_group(self, host, result_item):
         group_name = result_item.get('add_group')
         parent_group_names = result_item.get('parent_groups', [])
 
+        changed = False
```
Comment (tedder) [inline lib/ansible/plugins/strategy/__init__.py]: changed=false is defined about 5 lines up.

**Comment by maxamillion** [comment]
needs_revision

**Comment by winstonsimon** [comment]
I was hoping this is fixed in 2.5 atleast and Ansible is still broken for AWS users after 2.3.1.0

```ERROR! Unexpected Exception, this is probably a bug: u'localhost'
the full traceback was:

Traceback (most recent call last):
  File "/usr/bin/ansible-playbook", line 118, in <module>
    exit_code = cli.run()
  File "/usr/lib/python2.7/site-packages/ansible/cli/playbook.py", line 122, in run
    results = pbex.run()
  File "/usr/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 159, in run
    result = self._tqm.run(play=play)
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 290, in run
    play_return = strategy.run(iterator, play_context)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/linear.py", line 291, in run
    results += self._wait_on_pending_results(iterator)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 680, in _wait_on_pending_results
    results = self._process_pending_results(iterator)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 117, in inner
    results = func(self, iterator, one_pass=one_pass, max_passes=max_passes)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 592, in _process_pending_results
    self._add_group(original_host, result_item)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 726, in _add_group
    real_host = self._inventory.hosts[host.name]
KeyError: u'localhost'```


Ansible config:
```
- hosts: localhost
  connection: local

  tasks:
    - name: Load configuration for {{ envName }}_{{ region }}
      group_by: key="{{ envName }}_{{ region }}"
    - debug:
        msg: hello
```

**Comment by winstonsimon** [comment]
```[winstons@subtleshuttle ansible]$ ansible --version
ansible 2.5.0rc1
  config file = None
  configured module search path = [u'/winstons/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Apr  9 2015, 11:03:32) [GCC 4.8.3 20140911 (Red Hat 4.8.3-9)]```

**Comment by bcoca** [comment]
@winstonsimon you realize this has not been merged at all?

**Comment by winstonsimon** [comment]
If all good can you merge this and give us a new version? Issue is we are stuck at 2.3.1.0 and not able to use any new AWS features eg: NLB, Dynamodb etc.. 

This fix would be extremely helpful for us to move forward with Ansible.

**Comment by winstonsimon** [comment]
Thanks for adding it to blocker list. Meanwhile can you suggest some workaround to bypass this issue?

**Comment by winstonsimon** [comment]
When can we expect this change? Any ETA?

**Comment by bcoca** [comment]
@winstonsimon I don't think this relates to your issue at all, this is just adding 'changed state' to the `group_by` action. 

If you have a reproducer with your problem please open a ticket, fyi I have not heard any other AWS user report anything similar.

File: lib/ansible/inventory/data.py [hunk 2]
```diff
@@ -240,19 +240,20 @@ def set_variable(self, entity, varname, value):
 
     def add_child(self, group, child):
         ''' Add host or group to group '''
-
+        added = False
         if group in self.groups:
             g = self.groups[group]
             if child in self.groups:
-                g.add_child_group(self.groups[child])
+                added = g.add_child_group(self.groups[child])
```
Comment (maxamillion) [inline lib/ansible/inventory/data.py:261]: `add_child_group` doesn't appear to return anything, does this result in a non-falsey value?

Comment (bcoca) [reply lib/ansible/inventory/data.py:261]: it returns true/false if something was added or not

**Review by jimi-c — APPROVED** [review]

**Pull request merged**